src/librustc/middle/stability.rs

```diff
@@ -564,18 +564,20 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
             return EvalResult::Allow;
         }
 
-        let lint_deprecated = |def_id: DefId, id: NodeId, note: Option<Symbol>| {
-            let path = self.item_path_str(def_id);
-
+        let lint_deprecated = |def_id: DefId,
+                               id: NodeId,
+                               note: Option<Symbol>,
+                               message: &str,
```

Comment:
I wonder if it would be more appropriate to not pass the message in at all and instead to choose the message inside the closure by matching against `lint`

Reply:
The `Lint` just has a string identifier, rather than an enum, so we wouldn't get any assurances that this code would be updated if the lints ever changed, so I think passing it in is a better choice here.